### PR TITLE
Feature: links open above or below selected tab

### LINKED ISSUE #462: Opening new tab with alt+enter opens the tab at the botton
In an existing tab with some page loaded, type something in the address bar and hit alt+enter. A new tab will open with that link but it will open at the bottom of the list.

### PR DESCRIPTION
**Pull request by ericawright**
reviewer: @bwinton

Override addTab method, move new tab and linked tab according to user pref.

fixes: #410.
fixes #462.

### CODE REVIEW

**Comment by bwinton** [comment]
Yeah, it doesn't seem to be working for me at all…  (Like, it throws an exception in nightly and devedition, and doesn't load any tabs or the browser toolbox, so I can't debug it.  😕)
I wonder if there's a different (better) way of doing this?

**Comment by ericawright** [comment]
yeah, I think this branch is better
https://github.com/ericawright/VerticalTabs/tree/method-override

File: verticaltabs.js [hunk 1]
```diff
@@ -87,12 +88,33 @@ VerticalTabs.prototype = {
     let window = this.window;
     let document = this.document;
 
-    this.BrowserOpenTab = this.window.BrowserOpenTab;
-    this.window.BrowserOpenTab = function () {
-      this.pushToTop = prefs.opentabstop;
-      this.window.openUILinkIn(this.window.BROWSER_NEW_TAB_URL, 'tab');
-      this.pushToTop = false;
-    }.bind(this);
+    let oldAddTab = window.gBrowser.addTab.bind(window.gBrowser);
```
Comment (bwinton) [inline verticaltabs.js]: So, I don't think we should bind this until we call it because…

File: verticaltabs.js [hunk 2]
```diff
@@ -87,12 +88,33 @@ VerticalTabs.prototype = {
...
+    window.gBrowser.addTab = function (...args) {
```
Comment (bwinton) [inline verticaltabs.js]: We should re-set `window.gBrowser.addTab` to the previous value in an unload function…

**Comment by bwinton** [comment]
Other than that, there's a small nit: Cmd-clicking a pinned tab doesn't scroll to the newly opened tab…  I'm okay with not fixing that, though.  🙂

File: verticaltabs.js (hunk 2 above)
Comment (ericawright) [inline verticaltabs.js]: yup, I always forget to reset

**Comment by bwinton** [comment]
One more nit: If I'm on the last tab, and open one above, it doesn't show the tab I'm on anymore…

**Comment by bwinton** [comment]
(But it's also ready for merging, once it's rebased…  🙂)

**Comment by ericawright** [comment]
yeah, I think both of those issues can be their own separate issues

**Pull request merged**